nipoppy/workflows/dataset_init.py

```diff
@@ -85,13 +107,65 @@ def run_main(self):
             FPATH_SAMPLE_MANIFEST, self.layout.fpath_manifest, log_level=logging.DEBUG
         )
 
+        if self.bids_source is not None:
+            self._init_manifest_from_bids_dataset()
+
         # inform user to edit the sample files
         self.logger.warning(
             f"Sample config and manifest files copied to {self.layout.fpath_config}"
             f" and {self.layout.fpath_manifest} respectively. They should be edited"
             " to match your dataset"
         )
```

Comment:
I think it would make sense to do something like:

```python
if self.bids_source is not None:
    self._init_manifest_from_bids_dataset()
else:
    self.copy(FPATH_SAMPLE_MANIFEST, self.layout.fpath_manifest, log_level=logging.DEBUG)
```

to avoid the unnecessary copying of the sample manifest that immediately gets overwritten.

Also we might want to split the warning onto one for the global config and one for the manifest so that if `--bids-source` is specified there is only a warning about the global config? Not sure